client/app/components/DiningHallItem.js

```diff
@@ -71,20 +71,49 @@ export default class DiningHallItem extends Component {
 
     isOpenStyle = () => this.props.isOpen ? { opacity: 1 } : { opacity: 0.9 }
 
+    imageSource = () => {
+        switch (this.props.name) {
+            case "Berkeley": return require("../../assets/images/Berkeley.png");
+            case "Branford": return require("../../assets/images/Branford.png");
+            case "Grace Hopper": return require("../../assets/images/Grace_Hopper.png");
+            case "Davenport": return require("../../assets/images/Davenport.png");
+            case "Jonathan Edwards": return require("../../assets/images/Jonathan_Edwards.png");
+            case "Morse": return require("../../assets/images/Morse.png");
+            case "Pierson": return require("../../assets/images/Pierson.png");
+            case "Saybrook": return require("../../assets/images/Saybrook.png");
+            case "Silliman": return require("../../assets/images/Silliman.png");
+            case "Stiles": return require("../../assets/images/Ezra_Stiles.png");
+            case "Timothy Dwight": return require("../../assets/images/Timothy_Dwight.png");
+            case "Trumbull": return require("../../assets/images/Trumbull.png");
+            case "Pauli Murray": return require("../../assets/images/Pauli_Murray.png");
+            case "Franklin": return require("../../assets/images/Benjamin_Franklin.png");
+            default: return require("../../assets/images/Yale_College.png")
+        }
+    }
+
     render() {
         return (
             <Transition appear="fade">
                 <View style={this.isOpenStyle()}>
                     <View style={{ ...styles.container.flexRow, ...styles.container.spaceBelow }}>
-                        <View style={{ width: '50%' }}>
+                        <View style={{ width: '10%' }}>
+                            <Image style={{ width: 30, height: 37 }} source={this.imageSource()} />
```

Comment:
Is there a better way to "flex size" this? Or is setting manual values the way to go? As you can tell, they were chosen by trial and error...

Reply:
Yeah, usually for setting image sizes, it's good to just modify the `width` style and make it a percentage. This will scale the image proportionally to its original size (height included) and using percentages for these image sizes will make the image scale correctly on different mobile device screens.

Reply:
Hm, apparently `Image` doesn't autoscale in react, so setting width only really crops the image. Setting both width and height to the same percentage makes each element really tall for some reason.

Reply:
Ah, well apparently you need to set `resizeMode=contain` as one of the props! According to [this stackoverflow post](https://stackoverflow.com/questions/42170127/auto-scale-image-height-with-react-native). Did that work?

Reply:
Hm, the logo resizes correctly but there's still the issue of adding excessive padding on the top and bottom of each element.

Reply:
You're saying there's too much padding when you use `resizeMode=contain`? What are your style parameters now?

Reply:
```
<View style={{ width: '10%' }}>
    <Image resizeMode='contain' style={{ width: '100%' }} source={this.imageSource()} />
</View>
```
Which displays:
![80158](https://user-images.githubusercontent.com/18409216/55262055-044dba00-5243-11e9-9eeb-4f3e7ea898bf.jpg)

Reply:
Hey Zeb. Yeah you're right.

I took a look at [this article...](https://medium.com/the-react-native-log/tips-for-react-native-images-or-saying-goodbye-to-trial-and-error-b2baaf0a1a4d) Try doing this!:

```jsx
<View style={{ width: '10%'}}>
    <Image style={{flex: 1, width: undefined, height: undefined}} 
            resizeMode='contain' 
            source={this.imageSource()} 
    />
</View>
```

Reply:
Flexing on me, I see :)